examples/deep_learning/distributed_flax_imagenet.py

```diff
@@ -0,0 +1,839 @@
+# Copyright 2022 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+SPMD ResNet example with Flax and JAXopt.
+====================================
+"""
+
+import functools
+import itertools
+import os
+import time
+from typing import Any, Callable, Iterator, Mapping, NamedTuple, Optional, Sequence, Type, Tuple, Union
+
+from absl import app
+from absl import flags
+from absl import logging
+
+from chex import Array, ArrayTree, Numeric, PRNGKey
+
+from clu import checkpoint
+from clu import metric_writers
+from clu import metrics as clu_metrics
+from clu import periodic_actions
+
+from flax import linen as nn
+from flax import struct
+
+import jax
+from jax import numpy as jnp
+from jax import random
+from jax.experimental.maps import Mesh
+from jax.experimental.pjit import PartitionSpec
+from jax.experimental.pjit import pjit
+
+import jaxopt
+from jaxopt import loss as jaxopt_losses
```

Comment:
I would just use `jaxopt.loss` directly where you need it instead.

Reply:
SGTM, done!